src/Server.java

```diff
@@ -1,2 +1,53 @@
-public class Server {
+import java.io.IOException;
+import java.net.*;
+
+public class Server extends Thread {
+    private final int PORT = 9876;
+    private DatagramSocket serverSocket;
+    private byte[] buf = new byte[256];
+    private boolean running;
+
+    public Server() {
+        System.out.println("Server()");
+
+        try {
+            serverSocket = new DatagramSocket(PORT);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void run() {
+        System.out.println("run()");
+        running = true;
+        while (running) {
+            DatagramPacket receivePacket = new DatagramPacket(buf, buf.length);
+            try {
+                serverSocket.receive(receivePacket);
+                System.out.println("received packet from client");
+
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            String receivedSentence = new String(receivePacket.getData(), 0, receivePacket.getLength());
+            InetAddress IPAddress = receivePacket.getAddress();
+            int port = receivePacket.getPort();
+
+            DatagramPacket sendPacket = new DatagramPacket(buf, buf.length, IPAddress, port);
+            try {
+                serverSocket.send(sendPacket);
+                System.out.println("packet sent to client");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            if (receivedSentence.equals("end")) {
```

Comment:
Let's pull "end" to public static final field so client can also use this field.

Reply:
okay